### PR TITLE
Add section on dotnet watch behavior with response compression

### LINKED ISSUE #31949: Update 'dotnet watch' doc to detail new .NET 7 behavior
@MackinnonBuck 

The browser refresh middleware used by `dotnet watch` and Visual Studio to auto-refresh the browser when code changes are detected was [updated in .NET 7](https://github.com/dotnet/sdk/pull/27537) to make it more resilient to failures under certain app configurations, including when [response compression is enabled](https://github.com/dotnet/aspnetcore/issues/32767).

In these cases it will log a warning to the application output like the following:
```
warn: Microsoft.AspNetCore.Watch.BrowserRefresh.BrowserRefreshMiddleware[4]
      Unable to configure browser refresh script injection on the response. This may have been caused by the response's Content-Encoding: 'br'. Consider disabling response compression.
```

We should update this doc (and perhaps the [VS Hot Reload docs](https://learn.microsoft.com/en-us/visualstudio/debugger/hot-reload?view=vs-2022#configure-hot-reload) too) with the details of this change, including that folks can manually add the browser refresh JavaScript reference to their app's pages to get browser refresh working, e.g.:

```
@if (Environment.GetEnvironmentVariable("__ASPNETCORE_BROWSER_TOOLS") is not null)
{
<script src="/_framework/aspnetcore-browser-refresh.js"></script>
}
```

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 16f49f1b-261c-8cd0-a024-32c2535d0a0f
* Version Independent ID: 218d2a4d-2ff3-8e4d-1f06-bbb3710878fe
* Content: [dotnet watch command - .NET CLI](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-watch)
* Content Source: [docs/core/tools/dotnet-watch.md](https://github.com/dotnet/docs/blob/main/docs/core/tools/dotnet-watch.md)
* Product: **dotnet-fundamentals**
* Technology: **dotnet-cli**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

---
[Associated WorkItem - 34608](https://dev.azure.com/msft-skilling/Content/_workitems/edit/34608)

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #31949 

[Internal review URL](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-watch?branch=pr-en-us-32481)

### CODE REVIEW

File: docs/core/tools/dotnet-watch.md [hunk 1]
```diff
@@ -30,6 +30,23 @@ The `dotnet watch` command is a file watcher. When it detects a change that is s
...
 
+### Response compression
+
+If `dotnet watch` runs for an app that uses [response compression](/aspnet/core/performance/response-compression), the tool can't inject the browser refresh script. In that case, the .NET 7 version of the tool displays a warning message like the following:
+
+> warn: Microsoft.AspNetCore.Watch.BrowserRefresh.BrowserRefreshMiddleware[4]
+>
+> Unable to configure browser refresh script injection on the response. This may have been caused by the response's Content-Encoding: 'br'. Consider disabling response compression.
+
+As an alternative to disabling response compression, you can manually add the browser refresh JavaScript reference to the app's pages:
```
Comment (Rick-Anderson) [inline docs/core/tools/dotnet-watch.md:41]:
```suggestion
As an alternative to disabling response compression, manually add the browser refresh JavaScript reference to the app's pages:
```
per style guide, shorter is better.

**Review by Rick-Anderson — APPROVED** [review]
Why not put that in a moniker 7, that way you don't have to say .NET 7 and need to update it in .NET 8, 9, 10, 11

File: docs/core/tools/dotnet-watch.md [hunk 2]
```diff
@@ -30,6 +30,23 @@ The `dotnet watch` command is a file watcher. When it detects a change that is s
 
 While running `dotnet watch`, you can force the app to rebuild and restart by pressing Ctrl+R in the command shell. This feature is available only while the app is running. For example, if you run `dotnet watch` on a console app that ends before you press Ctrl+R, pressing Ctrl+R has no effect. However, in that case `dotnet watch` is still watching files and will restart the app if a file is updated.
 
+### Response compression
+
+If `dotnet watch` runs for an app that uses [response compression](/aspnet/core/performance/response-compression), the tool can't inject the browser refresh script. In that case, the .NET 7 version of the tool displays a warning message like the following:
```
Comment (Rick-Anderson) [inline docs/core/tools/dotnet-watch.md:35]:
```suggestion
If `dotnet watch` runs for an app that uses [response compression](/aspnet/core/performance/response-compression), the tool can't inject the browser refresh script. The  .NET 7 and later version of the tool displays a warning message like the following:
```

Comment (Rick-Anderson) [reply docs/core/tools/dotnet-watch.md:35]: Now you don't need a moniker.

**Comment by tdykstra** [comment]
> Why not put that in a moniker 7, that way you don't have to say .NET 7 and need to update it in .NET 8, 9, 10, 11

We don't do moniker blocks in this repo.

**Pull request merged**